File: modules/strategy/src/test/sample_data/inputs/send_message.rb
require 'java'
java_import org.marketcetera.strategy.ruby.Strategy
java_import org.marketcetera.quickfix.FIXVersion
java_import org.marketcetera.trade.BrokerID
java_import java.lang.Long
java_import java.util.Date
java_import 'quickfix.field.TransactTime'

class SendMessage < Strategy
    def on_start
        do_send
    end
    def on_stop
        do_send
    end
    def do_send
        messageDate = Long.parseLong(get_parameter("date"))

        nullMessage = get_parameter("nullMessage")
        if(nullMessage == nil)
            message = FIXVersion.getFIXVersion("FIX.0.0").getMessageFactory().newBasicOrder()
            message.setField(TransactTime.new(Date.new(messageDate)))
        else
            message = nil
        end

        nullBroker = get_parameter("nullBroker")
        if(nullBroker == nil)
            broker = BrokerID.new("some-broker")
        else
            broker = nil
        end
        
        send_message(message, broker)
    end  
end

File: size_change/modules/strategy/src/test/sample_data/inputs/SendMessage.java
import java.util.Date;

import org.marketcetera.core.quickfix.FIXVersion;
import org.marketcetera.strategy.java.Strategy;
import org.marketcetera.core.trade.BrokerID;

import quickfix.Message;
import quickfix.field.TransactTime;

/**
 * Test strategy to send messages via the FIX escape hatch.
 *
 * @version $Id: SendMessage.java 16063 2012-01-31 18:21:55Z colin $
 * @since 1.0.0
 */
public class SendMessage
        extends Strategy
{
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStart()
     */
    @Override
    public void onStart()
    {
        doSend();
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStop()
     */
    @Override
    public void onStop()
    {
        doSend();
    }
    /**
     * Sends messages according to strategy parameters.
     */
    private void doSend()
    {
        long messageDate = Long.parseLong(getParameter("date"));
        String nullMessage = getParameter("nullMessage");
        Message message;
        if(nullMessage == null) {
            message = FIXVersion.FIX_SYSTEM.getMessageFactory().newBasicOrder();
            message.setField(new TransactTime(new Date(messageDate)));
        } else {
            message = null;
        }
        String nullBroker = getParameter("nullBroker");
        BrokerID broker;
        if(nullBroker == null) {
            broker = new BrokerID("some-broker");
        } else {
            broker = null;
        }
        sendMessage(message,
                    broker);
        setProperty("onStart",
                    Long.toString(System.currentTimeMillis()));
    }
}
